src/test/java/uk/gov/companieshouse/api/accounts/controller/NoteControllerTest.java

```diff
@@ -0,0 +1,351 @@
+
+        ResponseEntity returnedResponse = noteController.get(COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, responseEntity.getStatusCode());
+        assertNull(responseEntity.getBody());
+    }
+
+    @Test
+    @DisplayName("Delete Note resource - success")
+    void deleteNoteResourceSuccess() throws DataException {
+
+        when(request.getAttribute(anyString())).thenReturn(transaction);
+        when(accountsNoteConverter.getAccountsNote(accountType, noteType)).thenReturn(accountingNoteTypeWithExplicitValidation);
+
+        ResponseObject<Note> responseObject = new ResponseObject<>(ResponseStatus.UPDATED,
+                note);
+
+        when(noteService.delete(accountingNoteTypeWithExplicitValidation, COMPANY_ACCOUNTS_ID, request))
+                .thenReturn(responseObject);
+
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.NO_CONTENT)
+                .build();
+        when(apiResponseMapper.map(responseObject.getStatus(), responseObject.getData(),
+                responseObject.getErrors()))
+                .thenReturn(responseEntity);
+
+        ResponseEntity returnedResponse =
+                noteController.delete(COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.NO_CONTENT, responseEntity.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Delete Note resource - data exception thrown")
+    void deleteNoteResourceDataException() throws DataException {
+
+        when(request.getAttribute(anyString())).thenReturn(transaction);
+        when(accountsNoteConverter.getAccountsNote(accountType, noteType)).thenReturn(accountingNoteTypeWithExplicitValidation);
+        when(noteService.delete(accountingNoteTypeWithExplicitValidation, COMPANY_ACCOUNTS_ID, request)).thenThrow(DataException.class);
+
+        ResponseEntity responseEntity =
+                ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+
+        when(apiResponseMapper.getErrorResponse()).thenReturn(responseEntity);
+
+        ResponseEntity returnedResponse = noteController.delete(COMPANY_ACCOUNTS_ID, accountType, noteType, request);
+
+        assertNotNull(returnedResponse);
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, responseEntity.getStatusCode());
+    }
+
+    @Test
+    @DisplayName("Tests the InitBinder for Note ")
+    void testInitDataBinderSuccess() {
+
+        AccountTypeConverter converter = new AccountTypeConverter();
+
+        AccountType type = AccountType.SMALL_FULL;
+        converter.setAsText("small-full");
+        WebDataBinder binder = new WebDataBinder(type, converter.getAsText());
+        noteController.initBinder(binder);
+        assertNotNull(binder);
+        assertEquals(type.toString(), converter.getAsText());
+    }
+}
```

Comment:
Would help to add verify at the end of each method to see if the methods are being called ?

Reply:
We could, and I'll add `verify`'s if you wish, though I'd say it's not overly important; since we've mocked the behaviour of the dependencies, we know our mocks are being called otherwise Mockito would throw an `UnnecessaryStubbingException`. So by virtue of the tests running successfully, we know the mocks execute.

Reply:
Yes please there are too many interactions would be helpful to see the verify